Add 0.99.8 release note (#2866)
line/armeria: site/src/pages/release-notes/0.99.8.mdx

                               trailers.getInt(GrpcHeaderNames.GRPC_STATUS) != 0;
                    });
                })))
        .build(MyGrpcStub.class);
  ```
- You can now access the metadata of <type://HttpFile> and <type://FileService> asynchronously. #1668 #2845
  ```java
  HttpFile httpFile = HttpFile.of(...);
  CompletableFuture<HttpFileAttributes> attrs =
      httpFile.readAttributes(CommonPools.blockingTaskExecutor());
  CompletableFuture<ResponseHeaders> headers =
      httpFile.readHeaders(CommonPools.blockingTaskExecutor());
  ```
- You can now easily customize <type://MeterIdPrefixFunction> using
  <type://MeterIdPrefixFunction#andThen(MeterIdPrefixFunctionCustomizer)>. #2839
  ```java
  MeterIdPrefixFunction
          .ofDefault("hoge")
          .andThen((registry, log, id) -> {
              return id.withTags("grpc-status", log.responseTrailers().get("grpc-status"));
          });
  ```
- Both synchronous and asynchronous APIs to select an <type://Endpoint> are added to
  <type://EndpointSelector> and thus <type://EndpointGroup> which is a sub-interface. #1910 #2837
  ```java
  ClientRequestContext ctx = ...
  EndpointGroup group = ...

>    // synchronous selection!

Reviewer: nit: ident with two spaces?
```suggestion
  // synchronous selection!
```

  Endpoint endpoint = group.selectNow(ctx);

>    // asynchronous selection!

Reviewer: Ditto.

  CompletableFuture<Endpoint> future = group.select(ctx, ctx.eventLoop(), 1000 /* timeout */ );
  ```
  - Thanks to this change, you can now avoid the <type://EmptyEndpointGroupException> at least
    for the case where the <typeplural://Endpoint> disappear temporarily.
- You can now get the snapshot status of circuit breakers from monitoring system. #2841
- You can now schedule a request timeout in nanoseconds. #2864
  - <type://ServiceRequestContext#setRequestTimeout(TimeoutMode,Duration)>

>   - <type://ClientRequestContext#setResponseTimeout(TimeoutMode,Duration)>

Reviewer: Move to `Improvements` section? Because the API is not changed at all.

## 📈 Improvements

- You can now see the error message by `ResourceLeakDetector` if the `DefaultClientFactory` is not closed
  properly. #2847
- The initial connection latency is lowered by giving up hopeless HTTP/1.1 connection early when
  HTTP/2 upgrade request fails. #2852

## 🛠️ Bug fixes

- Armeria server does not convert the `Host` header to `:authority` header anymore. #2846 #2850
  - <type://RequestHeaders#authority()> looks into `Host` header if `:authority` header is missing.
- You can retry when <type://DnsTimeoutException> is raised using <type://RetryRule#onUnprocessed()>. #2836

> - You no longer see `Http2Exception: failed to consume a HEADE` error. #2832

Reviewer: ```suggestion
- You no longer see `Http2Exception: failed to consume a HEADERS frame` error. #2832
```

- You no longer see `You tried to set the content preview twice` warning unless
  you really did apply content previewing decorator twice. #2830 #2832
- A non-base <type://WebClient> respects the user defined scheme correctly. #2835
- The specified port when building <type://EurekaUpdatingListener> and <type://ZooKeeperUpdatingListener>
  is used in the registration information of an Armeria server. #2826

## 🏚️ Deprecations

- All Spring registration beans are deprecated. Use <type://ArmeriaServerConfigurator> and
  <type://DocServiceConfigurator> instead. #2787 #2838
  - Deprecated beans: `HttpServiceRegistrationBean`, `AnnotatedServiceRegistrationBean`,

>     `GrpcServiceRegistrationBean` and `ThriftServiceRegistrationBean`

Reviewer: `MeterIdPrefixFunctionFactory` has been deprecated in favor of `RequestLog.serviceName()`. #2780 #2820

- `executor()`, `contextAwareExecutor()` and `contextAwareEventLoop()` methods in
  <type://RequestContext> are deprecated. #2834
- `on*(Predicate)` methods in the followings classes are deprecated. #2851
  - <type://RetryRule> and <type://RetryRuleBuilder>
  - <type://RetryRuleWithContent> and <type://RetryRuleWithContentBuilder>
  - <type://CircuitBreakerRule> and <type://CircuitBreakerRuleBuilder>
  - <type://CircuitBreakerRuleWithContent> and <type://CircuitBreakerRuleWithContentBuilder>
  - Use `on*(BiPredicate)` methods instead.

> - `RequestLogBuilder.defer*Content*()` and `isDefer*Content*Set()`` have been deprecated in favor of

Reviewer: ```suggestion
- `RequestLogBuilder.defer*Content*()` and `isDefer*Content*Set()` have been deprecated in favor of
```

  `defer(RequestLogProperty)` and `isDeferred(RequestLogProperty)`. #2832
- `MeterIdPrefixFunction.andThen(BiFunction<MeterRegistry, MeterIdPrefix, MeterIdPrefix>` is deprecated. #2839
- `UnprocessedRequestException(Throwable)` constructor is deprecated.

>   `Use <type://UnprocessedRequestException#of(Throwable)> instead. #2836

Reviewer: ```suggestion
  Use <type://UnprocessedRequestException#of(Throwable)> instead. #2836
```

  - An asynchronous <type://EndpointSelector#select(ClientRequestContext,ScheduledExecutorService,long)>
    has been added.
- <type://UserClient> API has been changed. #2837
  - Its constructor requires more parameters.
  - `execute()` requires less parameters.
- `StreamMessage.drainAll()` and its variants have been removed. #2782 #2827
  - They are only used for unit tests and you can use `reactor.test.StepVerifier` instead.
- `UnprocessedRequestException(String, Throwable)` constructor has been removed without replacement. #2836
- Deprecated timeout APIs in `{Service,Client}RequestContext` have been removed. #2864
  - `extend{Request,Response}TimeoutMillis(long)` and `extend{Request,Response}Timeout(Duration)`
  - `set{Request,Response}TimeoutAfterMillis(long)` and `set{Request,Response}TimeoutAfter(Duration)`
  - `set{Request,Response}TimeoutAtMillis(long)` and `set{Request,Response}TimeoutAt(Duration)`

## ⛓ Dependencies

- gRPC-Java 1.30.1 -> 1.30.2
- Jackson 2.11.0 -> 2.11.1
- Kafka 1.1.1 -> 2.5.0
- Micrometer 1.5.1 -> 1.5.2
- Reactor 3.3.6 -> 3.3.7
- Shaded dependencies
  - Caffeine 2.8.4 -> 2.8.5
- Examples:
  - Dagger 2.28 -> 2.28.1
  - monix_reactive 3.2.1 -> 3.2.2

>   - ScalaPB 0.10.1 -> 0.10.6

Reviewer: ScalaPB and monix_reactive are newly-introduced at this version. 😉

- Build:
  - Checkstyle 8.33 -> 8.34
  - Eureka 1.9.23 -> 1.9.25
  - Finagle ServerSets 20.5.0 -> 20.6.0
  - Gradle 6.5 -> 6.5.1
  - gRPC-Kotlin 0.1.3 -> 0.1.4

## 🙇 Thank you

<ThankYou
  usernames={[
    'anuraaga',

>     'heowc',

Reviewer: Add `@gary-lo` because of #2690

Reviewer: And `@hexoul` who reviewed PRs
